### ISSUE TITLE
Binder.validate for validating all bound fields

### ISSUE THREAD

**Issue by Artur-** [vaadin/framework8-issues#35]
```
// This will make all current validation errors visible
Set<BinderResult> validationErrors = binder.validate();

if (!validationErrors.isEmpty()) {
  Notification.show("Validation error count: "
    + validationErrors.size());
}
```

**Comment by hesara** [comment]
Currently just returns a Result<?>

**Comment by jdahlstrom** [comment]
8fcacd0a0afc5216d4d85898a28a39a846b66602 // basic validation
c674a979101001831391baa12dfebfdfef8e4278 // adds validate(), error reporting

**Comment by denis-anisimov** [comment]
Merged into #26
